src/ui/MLFacet/MLFacet.ts

```diff
@@ -17,18 +16,148 @@ import { Utils } from '../../utils/Utils';
 import { MODEL_EVENTS, IAttributesChangedEventArg } from '../../models/Model';
 import { Assert } from '../../misc/Assert';
 import { IFacetResponseValue } from '../../rest/Facet/FacetResponse';
+import { IResponsiveComponentOptions } from '../ResponsiveComponents/ResponsiveComponentsManager';
+import { IStringMap } from '../../rest/GenericParam';
+import { isFacetSortCriteria } from '../../rest/Facet/FacetSortCriteria';
+import { l } from '../../strings/Strings';
+
+export interface IMLFacetOptions extends IResponsiveComponentOptions {
+  id?: string;
+  title?: string;
+  field?: IFieldOption;
+  sortCriteria?: string;
+  numberOfValues?: number;
+  enableCollapse?: boolean;
+  collapsedByDefault?: boolean;
+  valueCaption?: any;
+}
 
+/**
+ * Renders a facet in the search interface.
+ *
+ * The main difference between this component and the `Facet` component
+ */
 export class MLFacet extends Component {
   static ID = 'MLFacet';
   static doExport = () => exportGlobally({ MLFacet });
-  static options: IMLFacetOptions = { ...MLFacetOptions, ...ResponsiveFacetOptions };
+
+  /**
+   * The options for the MLFacet
+   * @componentOptions
+   */
+  static options: IMLFacetOptions = {
+    ...ResponsiveFacetOptions,
```

Comment:
I left this spread here; it's not going to appear in the public docs, though.

Instead, perhaps the `IMLFacetOptions` interface should include the `enableResponsiveMode` and `dropdownHeaderLabel` properties. Then we could expose the documentation here. What do you guys think?

Reply:
This could work, but it does make the code more fragile, since we have to keep both places in sync, and could lead to bug otherwise.

Since it is not possible to document that using this method, if I have to choose between the two, the code being more "reliable" is more important than the documentation we generate with it.